tests/cases/fourslash/stringCompletionsFromGenericConditionalTypesUsingTemplateLiteralTypes.ts

```diff
@@ -28,4 +28,5 @@
 //// const shallowExpression = parse("foo|/*ts*/")
 //// const nestedExpression = parse({ prop: "foo|/*ts2*/" })
 
-verify.completions({ marker: ["ts", "ts2"], exact: ["foo|foo", "foo|bar", "foo|baz"] });
+verify.completions({ marker: ["ts"], exact: ["foo", "bar", "baz", "foo|foo", "foo|bar", "foo|baz"] });
```

Comment:
This matches Ron's initial expectations from https://github.com/microsoft/TypeScript/pull/54121 .

The other case, ideally, should provide the same completions but currently, the ones based on the contextual types and the ones based on `getCandidateSignaturesForStringLiteralCompletions` work quite differently (and even more so with changes from https://github.com/microsoft/TypeScript/pull/54121 ). I plan to close the gap between both a little bit but I'm leaving that for future PRs.